doc/specs/stdlib_string_type.md

```diff
@@ -0,0 +1,1607 @@
+- `pos`: Integer of default type. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a scalar character value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+character(len=:), allocatable :: dlc
+character(len=1), allocatable :: chars(:)
+
+string = "Character sequence"
+dlc = char(string, 3)
+! dlc == "a"
+chars = char(string, [3, 5, 8, 12, 14, 15, 18])
+! chars == ["a", "a", "e", "e", "u", "e", "e"]
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Char function (range variant)
+
+#### Description
+
+Return the character sequence represented by the string.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):char(interface)]] (string, start, last)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Pure function.
+
+#### Argument
+
+- `string`: Instance of a `string_type`. This argument is `intent(in)`.
+- `start`: Integer of default type. This argument is `intent(in)`.
+- `last`: Integer of default type. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a scalar character value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+character(len=:), allocatable :: dlc
+
+string = "Fortran"
+dlc = char(string, 1, 4)
+! dlc == "Fort"
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Ichar function
+
+Character-to-integer conversion function.
+
+#### Description
+
+Returns the code for the character in the first character position of the
+character sequence in the system's native character set.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):ichar(interface)]] (string)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Instance of a `string_type`. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: code
+
+string = "Fortran"
+code = ichar(string)
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Iachar function
+
+Code in ASCII collating sequence.
+
+#### Description
+
+Returns the code for the ASCII character in the first character position of
+the character sequences represent by the string.
+
+#### Syntax
+
+`res = [[stdlib_string_type(module):iachar(interface)]] (string)`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Elemental function.
+
+#### Argument
+
+- `string`: Instance of a `string_type`. This argument is `intent(in)`.
+
+#### Result value
+
+The result is a default integer scalar value.
+
+#### Example
+
+```fortran
+use stdlib_string_type
+implicit none
+type(string_type) :: string
+integer :: code
+
+string = "Fortran"
+code = iachar(string)
+end
+```
+
+
+<!-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -->
+### Index function
+
+Position of a *substring* within a *string*.
```

Comment:
This is a repetiton of the description. I suggest to remove this line.

Reply:
Those short descriptions were meant as man page like synopsis while the description itself is more concrete on the actual functionality and argument qualifiers and as an attempt to fill the space between the function name header and the description header.

Reply:
Mmmm, if I look for a description of a procedure, I would go to the section "Description",  not before it. I just found that strange.

The idea for these specs was also to have a structure similar to the one describing the procedure in the [Fortran Standard](https://j3-fortran.org/doc/year/18/18-007r1.pdf). If we go for your proposition, all other specs should be adapted too, for consistency.